File: client/models/Announcement.ts
export interface Announcement {
  announcementId: number;
  roomId: number;
  createdBy: number;
  message: string;
  createdAt: string;
  memberName: string;
  canReply: boolean;
}

export interface AnnouncementCreateRequest {
  roomId: number;
  message: string;
  canReply: boolean;
}

File: api/src/models/announcement.py
from pydantic import BaseModel
from datetime import datetime

class AnnouncementCreateRequest(BaseModel):
    room_id: int
    message: str
    can_reply: bool = False

class Announcement(BaseModel):
    announcement_id: int
    room_id: int
    created_by: int
    message: str
    can_reply: bool
    created_at: datetime

class AnnouncementResponse(BaseModel):
    announcement_id: int
    room_id: int
    created_by: int
    message: str
    can_reply: bool
    created_at: datetime
    member_name: str  # Joined from user table
